wkcuber/convert_nifti.py

```diff
@@ -0,0 +1,177 @@
+import time
+import logging
+import wkw
+import numpy as np
+from argparse import ArgumentParser
+import os
+import nibabel as nib
+
+from .utils import (
+    add_verbose_flag,
+    open_wkw,
+    WkwDatasetInfo,
+    ensure_wkw,
+    add_distribution_flags,
+    setup_logging,
+)
+
+from .metadata import write_webknossos_metadata
+
+
+def create_parser():
+    parser = ArgumentParser()
+
+    parser.add_argument(
+        "source_path", help="Directory containing the source NIFTI dataset."
+    )
+
+    parser.add_argument(
+        "target_path", help="Output directory for the generated WKW dataset."
+    )
+
+    parser.add_argument(
+        "--layer_name",
+        "-l",
+        help="Name of the cubed layer (color or segmentation).",
+        default="color",
+    )
+
+    parser.add_argument(
+        "--dtype",
+        "-d",
+        help="Target datatype (e.g. uint8, uint16, uint32).",
+        default="uint8",
+    )
+
+    parser.add_argument(
+        "--convert_folder",
+        help="Bool indicating whether to or not to convert complete folder, default is false ",
+        type=bool,
+        default=False,
+    )
+
+    parser.add_argument(
+        "--color_file",
+        help="When converting folder, name of file to become color layer",
+        default=None,
+    )
+
+    parser.add_argument(
+        "--segmentation_file",
+        help="When converting folder, name of file to become segmentation layer",
+        default=None,
+    )
+
+    add_verbose_flag(parser)
+    add_distribution_flags(parser)
+
+    return parser
+
+
+def to_target_datatype(data: np.ndarray, target_dtype) -> np.ndarray:
+
+    if data.dtype == np.dtype("float32") or data.dtype == np.dtype("float64"):
+        factor = data.max()
+
+    else:
+        factor = np.iinfo(data.dtype).max
+
+    if target_dtype != np.dtype("float32"):
+        factor = factor / np.iinfo(target_dtype).max
```

Comment:
I find this logic a bit hard to grasp, since the calculation is spread over to ifs. How about merging them like this:

```python
    if data.dtype == np.dtype("float32"):
        factor = data.max()
    elif data.dtype == np.dtype("float64"):
        factor = data.max() / np.iinfo(target_dtype).max
    else:
        factor = np.iinfo(data.dtype).max / np.iinfo(target_dtype).max
```

However, I'm not sure, whether this makes 100% sense. If the input is in float32 (and the output too), the output will be normalized between -1 and 1, right? However, there shouldn't be any change happening to the data from my understanding?

Reply:
You're right. I think the changes should do it.